Support for block ID

uroybd opened the pull request:
This PR complements https://github.com/oleeskild/obsidian-digital-garden/pull/141

oleeskild reviewed (changes requested):
Thanks!
There are a lot of changes in this PR that are formatting changes. 
As far as I can see the only real changes are the css for div.transclsion, and the addition of the markdown-it-attrs plugin. Could you clean it up a bit so only the real changes are committed?
Also the Test Doc.md needs to be removed :)

uroybd commented:
Hi, removed the superfluous formatting changes. However, also implemented backlinking for embedded contents. In the process, moved the repeated backlinking code in a helper utility, and refactored it a bit. I hope you don't mind.

oleeskild commented:
Great! Very nice addition and thanks for the refactor! That was definitely needed. 

This is ready to merge. Just waiting for the last fix in https://github.com/oleeskild/obsidian-digital-garden/pull/141 . Then they can be released simultaneously.
